api/src/wfl/module/covid.clj

```diff
@@ -93,6 +93,98 @@
   "Use `tx` to load the workload sink with `sink_type`."
   (fn [tx workload] (:sink_type workload)))
 
+;; validations
+(defmulti throw-when-malformed-source-request!
+  "Validate the source of a `request`"
+  (fn [source] (:name source)))
+
+(defmulti throw-when-malformed-executor-request!
+  "Validate the executor of a `request`"
+  (fn [executor] (:name executor)))
+
+(defmulti throw-when-malformed-sink-request!
+  "Validate the sink of a `request`"
+  (fn [sink] (:name sink)))
+
+;; Workload Functions
+(def ^:private dataset-table-name "flowcells")
+(def ^:private dataset-column-name "run_date")
+(def ^:private source-name "Terra DataRepo")
+(def ^:private executor-name "Terra")
+(def ^:private sink-name "Terra Workspace")
```

Comment:
Why are these here? They're already defined below for each of the source/sink/executor implementations.

Reply:
Some of these are parameters in the workload-request so please don't hard-code!

Reply:
I don't understand. Am I not supposed to validate that the names of these things are what we expect? If I do not use these strings, then where do I get the strings to validate against?

Reply:
The validation step occurs in the create-covid-workload, which is above the source/executor/sink sections, so I cannot refer to the variables in those sections. I moved the workload section under them and then was asked to move the section back above...